### ISSUE TITLE
linter/plugins: Handle/disable rule tests using impliedStrict

### ISSUE THREAD

**Issue by connorshea** [oxc-project/oxc#16862]
Many of the failures in the plugin conformance snapshot appear to be due to the `impliedStrict: true` option being set:

For example,

#### strict > valid

```js
function foo() { return; }
```

```json
{
  "languageOptions": {
    "parserOptions": {
      "ecmaFeatures": {
        "impliedStrict": true
      }
    }
  }
}
```

```
AssertionError [ERR_ASSERTION]: Should have no errors but had 1: [
  {
    ruleId: 'rule-to-test/strict',
    message: "Use the function form of 'use strict'.",
    messageId: 'function',
    severity: 1,
    nodeType: 'FunctionDeclaration',
    line: 1,
    column: 0,
    endLine: 1,
    endColumn: 26,
    suggestions: null
  }
]
```

```
1 !== 0

    at assertErrorCountIsCorrect (apps/oxlint/dist/index.js)
    at assertValidTestCasePasses (apps/oxlint/dist/index.js)
    at runValidTestCase (apps/oxlint/dist/index.js)
    at apps/oxlint/dist/index.js
```

This needs to be either implemented properly, or these tests should be disabled. However, it is plausible that these failures are related to the scope manager instead?

Docs for impliedStrict: https://eslint.org/docs/latest/use/configure/language-options#specifying-parser-options

**Comment by overlookmotel** [comment]
Thanks for discovering this. Do you have a guess how many tests are failing because of this?

Oxc parser doesn't have such an option. But maybe that doesn't matter because we ignore non-fatal parse errors anyway in conformance tests. Maybe you're right that the "strictness" info that the rule is acting on comes from scope manager.

**Comment by connorshea** [comment]
Looks like it's only 12 tests currently on main, from searching the conformance snapshot.

**Comment by overlookmotel** [comment]
Implemented in #18307.
